App/Extensions/UIKit.swift

```diff
@@ -140,16 +140,25 @@ extension UIFont {
         - textStyle: The base style for the returned font.
         - fontName: An optional font name. If nil (the default), returns the system font.
         - sizeAdjustment: A positive or negative adjustment to apply to the text style's font size. The default is 0.
+        - weight: A positive or negative adjustment to apply to the text style's font size. The default is 0.
     - returns:
         A font associated with the text style, scaled for the user's Dynamic Type settings, in the requested font family.
     **/
-    class func preferredFontForTextStyle(_ textStyle: TypedTextStyle, fontName: String? = nil, sizeAdjustment: CGFloat = 0) -> UIFont {
-        let descriptor = UIFontDescriptor.preferredFontDescriptor(withTextStyle: UIFont.TextStyle(rawValue: textStyle.UIKitRawValue))
+    class func preferredFontForTextStyle(_ textStyle: TextStyle, fontName: String? = nil, sizeAdjustment: CGFloat = 0, weight: UIFont.Weight) -> UIFont {
+        let descriptor = UIFontDescriptor.preferredFontDescriptor(withTextStyle: textStyle)
+        let metrics = UIFontMetrics(forTextStyle: textStyle)
+        var font = metrics.scaledFont(for: UIFont.systemFont(ofSize: descriptor.pointSize + sizeAdjustment, weight: weight))
+        
         if let fontName = fontName {
             return UIFont(name: fontName, size: descriptor.pointSize + sizeAdjustment)!
         } else {
-            return UIFont(descriptor: descriptor, size: descriptor.pointSize + sizeAdjustment)
+            if let descriptor = UIFont.systemFont(ofSize: descriptor.pointSize + sizeAdjustment, weight: weight).fontDescriptor.withDesign(.rounded) {
+                if Theme.defaultTheme().roundedFonts {
```

Comment:
The theme should probably be passed in by the caller, so we can support cases where e.g. the posts view theme is different from the default theme.

Totally open to other ways to get the theme in, since I know this method has a lot going on already.

Reply:
I see what you mean. When I actually port the spankykong themes in, I'll implement this. As it stands, no theme exists that changes the font based on the new rounded property and the default is No. Keen to close stuff out!